File: Sortings/PythonSortings/QuadraticSortings.py
def InsertSort(arr):
    """Функция сортировки вставками"""
    for pos in range(1, len(arr)):
        i = pos
        while i > 0 and arr[i] < arr[i-1]:
            arr[i], arr[i-1] = arr[i-1], arr[i]
            i-=1



def ChoiseSort(arr):
    """Функция сортировки выбором"""
    for pos in range(len(arr)-1):
        for x in range(pos,len(arr)):
            if arr[pos] > arr[x]:
                arr[pos],arr[x] = arr[x], arr[pos]

    

def BubbleSort(arr):
    """Функция сортировки пузырьком"""
    for pos in range(len(arr)-1):
        for x in range(0, len(arr) - 1 - pos):
            if arr[x]>arr[x+1]:
                arr[x],arr[x+1] = arr[x+1], arr[x]

File: QuickSort/QuickSort/QuadraticSortings.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharpSortings
{
    public static class QuadraticSortings
    {
        public static void BubbleSort(List<int> arr)
        {
            for(int i = 0; i < arr.Count; i++)
            {
                bool flag = false;
                for(int j = 1; j < arr.Count - i; j++)
                {
                    if(arr[j] < arr[j - 1])
                    {
                        (arr[j], arr[j-1]) = (arr[j-1], arr[j]);
                        flag = true;
                    }
                }
                if(!flag)
                    break;
            }
        }

        public static void InsertSort(List<int> arr)
        {
            int tmp;
            //Считаем что первый элемент отсортирован
            for(int i = 1; i < arr.Count; i++)
            {
                for(int j = i; j > 0; j--) //Идем вдоль строя и смотрим не нужно ли вставить элемент в отсортированную область
                {
                    if(arr[j] < arr[j - 1]) //Если анализируемый предыдущий эелемент больше текущего - меняем местами
                    {
                        tmp = arr[j - 1];
                        arr[j - 1] = arr[j];
                        arr[j] = tmp;
                    }
                    else
                        break;
                }
            }
        }

        public static void ChoiseSort(List<int> arr)
        {
            for(int i = 0; i < arr.Count - 1; i++)
            {
                for(int j = i + 1; j < arr.Count; j++)
                {
                    if(arr[j] < arr[i])
                    {
                        int tmp = arr[i];
                        arr[i] = arr[j];
                        arr[j] = tmp;
                    }
                }
            }
        }
    }
}
